### PR TITLE
[ticket/13795] Modify core event in posting to include poll data

### PR DESCRIPTION
**Pull request by javiexin**
Modified the core.posting_modify_template_vars event in posting.php
to extend the content of the page_data array to include poll data.
That required moving the event and merging the new set of template vars.

PHPBB3-13795

### CODE REVIEW

File: phpBB/posting.php [hunk 1]
```diff
@@ -1850,29 +1877,6 @@
...
 $template->assign_vars($page_data);
 
-// Build custom bbcodes array
-display_custom_bbcodes();
-
-// Poll entry
-if (($mode == 'post' || ($mode == 'edit' && $post_id == $post_data['topic_first_post_id']/* && (!$post_data['poll_last_vote'] || $auth->acl_get('m_edit', $forum_id))*/))
-	&& $auth->acl_get('f_poll', $forum_id))
-{
-	$template->assign_vars(array(
```
Comment (Elsensee) [inline phpBB/posting.php]: The array is now never assigned to the template.

Comment (javiexin) [inline phpBB/posting.php]: Line 1878 of the new file.

Comment (Elsensee) [inline phpBB/posting.php]: Oh, I'm so blind.. I'm so sorry! xD

File: phpBB/posting.php [hunk 2]
```diff
@@ -1238,6 +1238,7 @@
 	*
 	* @event core.posting_modify_submission_errors
 	* @var	array	post_data	Array with post data
+	* @var	array	poll		Array with poll data from post (must be used instead of the post_data equivalent)
```
Comment (nickvergessen) [inline phpBB/posting.php]: please add `@changed` to the event, see https://github.com/phpbb/phpbb/blob/740be0507e209afda5eb954ab674e557c2d8f8f6/phpBB/includes/ucp/ucp_prefs.php#L72

File: phpBB/posting.php [hunk 3]
```diff
@@ -1822,6 +1848,7 @@
 *		post_id, topic_id, forum_id, submit, preview, save, load,
 *		delete, cancel, refresh, error, page_data, message_parser
 * @change 3.1.2-RC1 Removed 'delete' var as it does not exist
+* @change 3.1.4-RC1 Added poll variables to the page_data array
```
Comment (nickvergessen) [inline phpBB/posting.php]: 3.1.5

File: phpBB/posting.php (hunk 2 above)
Comment (javiexin) [inline phpBB/posting.php]: Ok, forgot it here, while I did it in the below case.  Will send a new commit with these changes.

**Comment by nickvergessen** [comment]
Merged via https://github.com/phpbb/phpbb/commit/51831924c5bbd7ea63329e0455043a1a3c792837

**Pull request closed without merging**